Blog App -validations and model specs.

khanh-devos opened the pull request:
### Add the following validations:
#### For the User model:
- [x] Name must not be blank.
- [x] PostsCounter must be an integer greater than or equal to zero.
#### For the Post model:
- [x] Title must not be blank.
- [x] Title must not exceed 250 characters.
- [x] CommentsCounter must be an integer greater than or equal to zero.
- [x] LikesCounter must be an integer greater than or equal to zero.
- [x] Add unit specs for all of your models' methods and validations.

khanh-devos committed:
Add validation and unit test
- User: name not nil, postCounter: integer and >= 0
- Post: title not nil and max length 250, commentsCounter >= 0, likesCounter >= 0
- Test by gem rspec-rails (some at default rails test)

khanh-devos committed:
Fix rubocop linter errors

abdulhamiid commented on spec/test_user.rb:
```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+require 'rails_helper'
+
+RSpec.describe User do
+  context 'Text class User: ' do
+    before :each do
+      @name = 'Khanh'
+      @bio = 'Saigon'
+      @user = User.new(name: @name, bio: @bio)
+    end
+
+    it 'test all attrs' do
+      expect(@user.name).to eql(@name)
+      expect(@user.bio).to eql(@bio)
+      expect(@user.postCounter).to eql(nil)
+    end
+
+    it 'validate invalid blank name' do
+      user1 = User.new(name: nil, bio: 'bio', postCounter: 1)
+      expect(user1).to be_invalid
+    end
+
+    it 'validate postCounter greater_than_or_equal_to 0' do
+      user2 = User.new(name: @name, bio: 'bio', postCounter: -1)
+      expect(user2).to be_invalid
+    end
```
Nice job on the tests you have added 👍 

However, according to the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md) 🤗

- Please add unit specs for all of your models' methods and validations.
example
  - recent_3_posts in `user.rb`
  - recent_5_comments `post.rb` e.t.c
- So as to validate that each unit of the software works as intended and meets the requirements.

khanh-devos replied:
Sorry @abdulhamiid ,

abdulhamiid commented on spec/test_like.rb:
```diff
@@ -0,0 +1,14 @@
+require 'spec_helper'
+
+RSpec.describe Like do
+  context 'Test class Like: ' do
+    before :each do
+      @like = Like.new(user_id: 1, post_id: 2)
+    end
+
+    it 'test attrs' do
+      expect(@like.user_id).to eql(1)
+      expect(@like.post_id).to eql(2)
+    end
+  end
+end
```
Please fix the error in these 2 of your files to avoid a loading error
- `test_lik.rb` and `test_coment.rb`

![image](https://github.com/khanh-devos/rail-blog-app/assets/78251927/11432fea-8e77-40eb-a5b0-af779524650f)

You can fix that by adding [rails_helper ](https://medium.com/@brett_codes/rspecs-spec-helper-vs-rails-helper-explained-2337282d14ce)
```ruby
require 'rails_helper'

RSpec.describe Comment do
  before :each do
    @text = 'text'
    @comment = Comment.new(text: @text)
  end

  it 'test attrs' do
    expect(@comment.text).to eql(@text)
  end
end
```
```

khanh-devos replied:
This is a very great idea @abdulhamiid

abdulhamiid reviewed (changes requested):
# Status: Required Changes :recycle:
Hi @khanh-devos 👋, 
![](https://media.giphy.com/media/UdaSUYClr92cjWpeVx/giphy.gif)
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Use validations for models.
- ✔️ Write unit tests for models.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider the following:
- x
- y
- z

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

khanh-devos committed:
Fix test rspec following code review
- Test methods: recent_posts and recent_comments
- Test auto increment of postCounter, commentsCounter

t-yanick reviewed (approved):
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @khanh-devos , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested Changes implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
